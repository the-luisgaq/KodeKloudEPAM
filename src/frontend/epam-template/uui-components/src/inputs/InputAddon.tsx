import React from 'react';
import { cx, IHasCX } from '@epam/uui-core';
import css from './InputAddon.module.scss';

export interface InputAddonProps extends IHasCX {
    /** Content to be rendered in addon */
    content: React.ReactNode;
}

export function InputAddon(props: InputAddonProps) {
    return (
        <div className={ cx(css.prefixWrapper, props.cx, 'uui-input-addon') }>
            { props.content }
        </div>
    );
}
