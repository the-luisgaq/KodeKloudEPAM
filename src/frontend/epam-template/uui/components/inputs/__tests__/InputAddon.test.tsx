import React from 'react';
import { render } from '@epam/uui-test-utils';
import { InputAddon } from '../InputAddon';

describe('InputAddon', () => {
    it('should be rendered correctly', () => {
        const { asFragment } = render(<InputAddon content="Test Content" cx="custom-class" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
