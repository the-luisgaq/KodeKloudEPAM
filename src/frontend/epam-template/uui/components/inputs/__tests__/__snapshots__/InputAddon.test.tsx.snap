// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAddon should be rendered correctly 1`] = `
<DocumentFragment>
  <div
    class="prefixWrapper root custom-class uui-input-addon"
  >
    Test Content
  </div>
</DocumentFragment>
`;
